analysis/ic50/__init__.py

```diff
@@ -0,0 +1,90 @@
+from typing import Dict, List, Tuple, Union
+
+import numpy as np
+from scipy.optimize import curve_fit
+from sklearn.metrics import mean_squared_error
+from sklearn.preprocessing import MinMaxScaler
+
+
+def _sigmoid(x: Union[np.array, float], a: float, b: float, c: float, d: float) -> Union[np.array, float]:
+    return ((a - d) / (1.0 + ((x / c) ** b))) + d
+
+
+def _get_starting_parameters(concentrations: np.array, scaled_measurements: np.array) -> List[float]:
+    a = scaled_measurements[0]
+    d = scaled_measurements[-1]
+    b = 1
+    c = concentrations[(np.abs(scaled_measurements - (a + d) / 2)).argmin()]
+    return [a, b, c, d]
+
+
+def _fit_sigmoid(
+        measurements: np.array,
+        scaler: MinMaxScaler, fit_scaler: bool = True,
+) -> Tuple[np.array, np.array, float, MinMaxScaler]:
+    if fit_scaler:
+        scaled_signal = scaler.fit_transform(measurements[:, 1].reshape(-1, 1)).reshape(-1)
```

Comment:
How I like this reshapes))) Isn't there any better solution in python?

Reply:
That's so common for all ML libraries (in PyTorch you use reshapes all the time for matrices) that I didn't even think of it))))